wok/core/base.py

```diff
@@ -128,3 +129,35 @@ def finish(repo: pygit2.Repository, branch_name: str, message: str) -> None:
 
 def tag(repo: pygit2.Repository, tag_name: str) -> None:
     repo.references.create(name=f'refs/tags/{tag_name}', target=repo.head.target)
+
+
+class RemoteCallbacks(pygit2.RemoteCallbacks):
+    """
+    Overrides the credentials callback.
+    """
+
+    def credentials(
+        self, url: str, username_from_url: typing.Optional[str], allowed_types: int
+    ) -> typing.Optional[
+        typing.Union[pygit2.Keypair, pygit2.UserPass, pygit2.Username]
+    ]:
+        """
+        If the remote server requires authentication, this function will be called and
+        its return value used for authentication.
+        """
+        if allowed_types & pygit2.credentials.GIT_CREDTYPE_SSH_KEY:
+            return pygit2.KeypairFromAgent(username_from_url)
+        elif allowed_types & pygit2.credentials.GIT_CREDTYPE_USERPASS_PLAINTEXT:
+            username = username_from_url or input("Username: ")
+            password = getpass.getpass()
```

Comment:
I'm tempted to use `click`'s password prompt here. However, I see no real reason for that. So, just a thought.

Reply:
Agree with you. Addressed in 741facc.